### PR TITLE
Enhancement: Turn TalkController into CreateProcessAction

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] turns the `TalkController` into a `CreateProcessAction`

### CODE REVIEW

File: classes/Http/Action/Talk/CreateProcessAction.php [hunk 1]
```diff
@@ -133,64 +129,76 @@ public function processCreateAction(Request $request): Response
             'sponsor'     => $request->get('sponsor'),
             'user_id'     => $request->get('user_id'),
         ]);
-        $form->sanitize();
 
-        if ($form->validateAll()) {
```
Comment (localheinz) [inline classes/Http/Action/Talk/CreateProcessAction.php]: Inverted the condition to make it more obvious what the happy path is.

File: classes/Http/Action/Talk/CreateProcessAction.php [hunk 2]
```diff
@@ -88,40 +97,27 @@ public function __construct(
...
 
-            return $this->redirectTo('dashboard');
+            $url = $this->urlGenerator->generate('dashboard');
+
+            return new HttpFoundation\RedirectResponse($url);
         }
 
         $user = $this->authentication->user();
 
-        $form = $this->getTalkForm([
```
Comment (localheinz) [inline classes/Http/Action/Talk/CreateProcessAction.php]: Renamed this method as it actually creates a form.

File: classes/Http/Action/Talk/CreateProcessAction.php [hunk 3]
```diff
@@ -88,40 +97,27 @@ public function __construct(
...
+        $form = $this->createTalkForm([
```
Comment (localheinz) [inline classes/Http/Action/Talk/CreateProcessAction.php]: As the form is created here, we can't write unit tests that go further than the condition above.

**Comment by BackEndTea** [comment]
Currently trying to submit a talk on your branch gives an ```Swift_TransportException``` When submitting a talk. However this also occurs on master, so i'm not sure if this is my fault or that master is broken

**Comment by localheinz** [comment]
@BackEndTea 

Taking a look!

**Comment by localheinz** [comment]
@BackEndTea 

I could reproduce the issue in `master`, but it can be fixed by properly configuring the mail adapter.

**Comment by BackEndTea** [comment]
@localheinz 

I don't think that an error like this is intended behavior though. For example, how will the application respond if the email server is down?

**Comment by localheinz** [comment]
@BackEndTea 

Ideally, emails should be send asynchronously, but I think this is out of scope of this PR and should be handled separately.

**Comment by localheinz** [comment]
Thank you, @BackEndTea!

**Pull request merged**